### ISSUE TITLE
REST: Couldn't add "Linked Release" when add releases to components in API

### ISSUE THREAD

**Issue by chanrai** [eclipse/sw360#863]
Like a screen capture to attach, I want to add "Linked Release" information when add releases to components, and this could show the hierarchical structure.

Command will be the following.

$ curl -i -X POST -H "Authorization: Bearer <access_token>" -H "Content-Type: application/json" 'http://localhost:8080/resource/api/releases/<releases_ID>/' -d '[ ... "releaseIdToRelationship": {} ]'

I don't really understand how to write the "releaseIdToRelationship": {}.

I tried "releaseIdToRelationship": {
        "componentId":"d3cbcf1eb3855529d1287e87f906071e",
        "relationship":"Contained“
}

but it didn't work. 

Is API provided by this function provided?

Thank you.

![LinkedRelease](https://user-images.githubusercontent.com/35596645/82876220-68692c80-9f73-11ea-8a1b-b608d5efa98a.jpg)

**Comment by mcjaeger** [comment]
Yes, it is ... hve you seen the section "Link Releases to the project" in the API documentation? (hostname.com/resource/docs/api-guide.html)

**Comment by chanrai** [comment]
hi mcjaeger

Thank you for your reply.

Yes, I had read the API documentation about "Link Releases to the project", and It was confirmed by the API that the release id could links to the project. 

However, like the attached screen capture, I want to specify the hierarchical structure between components, so I think it is necessary to specify it in Release by Linked Release.

Is there any API that realizes that function?

Thank you.

![ReleaseHierachy](https://user-images.githubusercontent.com/35596645/83084178-35ce4980-a0c3-11ea-891c-3ec24007017d.jpg)

**Comment by chanrai** [comment]
This is a supplementary information.

When execute the json file with the following command.

`$curl --noproxy "127.0.0.1" 'http://127.0.0.1:8080/resource/api/releases' -i -X POST -H "Authorization: Bearer <API token>" -H "Content-Type:application/json" --data @"release.json"`

release.json
`{
 "createBy":"test@liferay.com",
 "type":"release",
 "cpeId":"",
 "name":"A Java library for reading/writing Excel",
 "version":"1.0",
 "componentId":"96fe00cb723eba51ab426401153b4974",
 "mainlineState":"OPEN",
 "releaseIdToRelationship":{"96fe00cb723eba51ab426401153b76b1":0 },
 "mainLicenseIds": ["MIT","0BSD"],
 "roles": {},
}`

Except the parameter `"releaseIdToRelationship"`, all parameters are recognized and the content have been registered on the SW360 side.

Check the log I found that just `"releaseIdToRelationship":{}`  also couldn't be recognized.

I also tried adding a parameter called `"testtest": ""` that shouldn't exist, but I was able to execute it without error.

So I think that the parameter `"releaseIdToRelationship"` couldn't be checked and be recognize in the API.

In the SW360 code, which file writes the parameter definition?

Such as `"ibraries/lib-datahandler/src/main/thrift/components.thrift"`, the definition about `"releaseIdToRelationship"` had be written.

I'm not sure why it doesn't work in the API.

**Comment by KoukiHama** [comment]
I test it and don't get error, then catalina.out showed

```
 2020-06-17 06:28:56 INFO  ReleaseController:195 - Debug1:
 2020-06-17 06:28:56 INFO  ReleaseController:196 - 96fe00cb723eba51ab426401153b4974
 2020-06-17 06:28:56 INFO  ReleaseController:197 - Release(type:release, cpeid:, name:A Java library fo  47588 r reading/writing Excel, version:1.0, componentId:96fe00cb723eba51ab426401153b4974, mainlineState:OPEN,  47588  roles:{}, mainLicenseIds:[0BSD, MIT])
```

**Comment by mcjaeger** [comment]
(from discussion) it would be good to have a separate endpoint just in the same way like for linking releases to projects.

note the backend is supporting it:
https://github.com/eclipse/sw360/blob/42c9c65524359854f8b126242b77e1bb8ed23b8d/libraries/lib-datahandler/src/main/thrift/components.thrift#L663

area for supporting it presumably could be here:
https://github.com/eclipse/sw360/tree/master/rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/release

**Comment by mcjaeger** [comment]
another hint from discussion as example how it could be done: https://github.com/eclipse/sw360/blob/42c9c65524359854f8b126242b77e1bb8ed23b8d/rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java#L198

**Comment by chanrai** [comment]
I found that `"releaseIdToRelationship"` will work if remove "releaseIdToRelationship" in https://github.com/eclipse/sw360/blob/master/rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java (Line 407)

Because releaseIdToRelationship is ignored in REST API.

After this fix, it works fine.

**Comment by KoukiHama** [comment]
Removing "releaseIdToRelationship" is good  temporary solution, but it should be fixed steadily.

**Comment by KoukiHama** [comment]
This issue was finally solved by #1310
